### PR TITLE
feat(pkger): add parser support for notification rules

### PR DESCRIPTION
**Pull request by jsteenb2**
- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: pkger/models.go [hunk 1]
```diff
@@ -107,7 +112,11 @@ func (k Kind) ResourceType() influxdb.ResourceType {
 }
 
 func (k Kind) title() string {
-	return strings.Title(k.String())
+	pieces := strings.Split(string(k), "_")
+	for i := range pieces {
+		pieces[i] = strings.Title(pieces[i])
```
Comment (jademcgough) [inline pkger/models.go]: heh, didn't know this was in strings

File: pkger/models.go [hunk 2]
```diff
@@ -922,18 +960,19 @@ func (c *check) valid() []validationErr {
...
-			Msg:   "must be 1 of [active, inactive]",
-		})
-	}
 	if c.statusMessage == "" {
 		vErrs = append(vErrs, validationErr{
 			Field: fieldCheckStatusMessageTemplate,
 			Msg:   `must provide a template; ex. "Check: ${ r._check_name } is: ${ r._level }"`,
 		})
 	}
+	if status := c.Status(); status != influxdb.Active && status != influxdb.Inactive {
```
Comment (jademcgough) [inline pkger/models.go]: demorgans law 👍

File: pkger/models.go [hunk 3]
```diff
@@ -1415,6 +1455,163 @@ func (n mapperNotificationEndpoints) Len() int {
...
+func (r *notificationRule) Name() string {
+	return r.name
+}
+
+func (r *notificationRule) ResourceType() influxdb.ResourceType {
+	return KindNotificationRule.ResourceType()
+}
+
+func (r *notificationRule) Status() influxdb.Status {
+	if r.status == "" {
```
Comment (jademcgough) [inline pkger/models.go]: When would status be unset for notification rules?

Comment (jsteenb2) [reply pkger/models.go]: good question. After speaking with @russorat, we decided that in pkger, you don't have to provide a status, it'll just default to `active` if not provided.

File: pkger/models.go (hunk 1 above)
Comment (jsteenb2) [reply pkger/models.go]: `strings` actually has a ton of useful funcs. Has stuff for `map`, 'trim`ing, and a boat load of other goodies that come up quite often

**Comment by jsteenb2** [comment]
tests failing upstream b/c of go vendor issue. This is not related to this PR work. Should not have any affect on any upstream stuffs, no new vendored code here, nor any changes to existing stuffs being used by idpe

File: pkger/models.go (hunk 3 above)
Comment (jademcgough) [reply pkger/models.go]: got it, that makes sense

**Pull request merged**